libcxx/include/deque

```diff
@@ -468,6 +475,13 @@ const _DiffType __deque_iterator<_ValueType, _Pointer, _Reference, _MapPointer,
 
 template <class _Tp, class _Allocator /*= allocator<_Tp>*/>
 class _LIBCPP_TEMPLATE_VIS deque {
+  static_assert(!is_const<_Tp>::value, "'std::deque' cannot hold const types");
```

Comment:
I would like to centralize these checks in some way, since they seem to all basically come from the same container requirements in the standard. If we ever change those requirements, it's almost certain that containers will get out of sync unless we centralize these checks.

Reply:
Working on this. Where is a good spot to put it? I was originally putting it in `__config`, but that's not great IMO. Possibly `__type_traits`, since it only consumes type traits?

Reply:
I think `__type_traits/diagnostic_utilities.h` or something like that would be good.